### ISSUE TITLE
How can it be that slow?

### ISSUE THREAD

**Issue by NoVa-UX** [omarroth/invidious#1138]
Does anybody else realize Invidious is pretty slow rn?  Or is it just me?  Honestly I use it all the time because school blocks certain videos that I NEED to watch and this is a great way to use it...I also like to listen to music and what not.  But now I'm always getting "500 internal server error" or the other title when I click on a video.  It can't be loaded?  What?  How?  Isn't this system supposed to be a backpage youtube?  Why doesn't it work like youtube then?  Why doesn't it work that much?  I get it..the system isn't perfect...nothing is.  But this?  This should be fixed sometime soon.  It's so slow that I'd rather use another site at this point.  Please...PLEASE  make it faster somehow...I really do enjoy this system.

**Comment by WandeingLagomorph793** [comment]
Can confirm, every other page is either a blank error page or a 500 internal server error, at least when I load it via the normal inivio.us, sometimes if I brute force it the page, and video, will load but it's still something that happens

**Comment by mendel5** [comment]
@NoVa-UX @WandeingLagomorph793 
Are you using the official instance at https://invidio.us/ ?

Maybe you can try other instances listed in https://github.com/omarroth/invidious/wiki/Invidious-Instances , e.g. https://invidious.snopyta.org/ . Are these instances also slow?

**Comment by NoVa-UX** [comment]
@mendel5 
I tried using those as well for me it is very slow.  I had to wait 10 minutes for it to load.  Too long.  And if I do get it to load.  "500 internal server error" and then I have to do it all over again.

**Comment by WandeingLagomorph793** [comment]
@mendel5 Official instance is super slow for me, just like the OP mentioned (The official invidio.us one), switched to Snopyta and that one was faster, at least as I write it now, not sure if it'll be slow later.

**Comment by me-ch** [comment]
I had the same issue with https://invidio.us but I tried the other ones and found out https://invidious.13ad.de to be the most optimal. I do not know why it worked because I am quite far from Germany(host location).

**Comment by NoVa-UX** [comment]
@me-ch 
It is still slow for me.  At first it was pretty fast and I was about to close this thread but then...that's when it started again.  And to people that are like "Oh it has to be your wifi then" its not.  I just paid the bill and it should be faster.  But the problem is no matter what.  I still cannot get it to go faster.

**Comment by NoVa-UX** [comment]
For example image down below is my subscriptions...its been loading for about a good 20 minutes...and STILL hasn't done anything
![image](https://user-images.githubusercontent.com/58235380/80655585-ae020a80-8a44-11ea-921f-8a6c64e7dbeb.png)

**Comment by me-ch** [comment]
It might be your location/cache/cookies?

**Comment by trymeouteh** [comment]
I find that invidio.us can be slow sometimes and I think it is because it is the most popular instance. We need to use other instances to spread the traffic around.

**Comment by elypter** [comment]
invidio.us should promote those instances or redirect to them for playback.

**Comment by me-ch** [comment]
try every single instance and see what works better I guess

**Comment by unixfox** [comment]
Bots are abusing the public instances.
After setting up an anti bot protection (sorry users with NoScript) my instance is running perfectly fine just like back in the days.

**Comment by mendel5** [comment]
@unixfox How do you know it is about the bots? I can think of two reasons for why the official Invidious instance ( https://invidio.us/ ) is not as performant as yours ( https://yewtu.be/ ).

1. Available computing resources
If we assume that all Invidious instances have the same number of users/ amount of load, then the provisioned computing resources can limit the performance. Maybe your instance has more CPU power, RAM, etc. than other instances?

2. Number of users
In (1) I assumed that all Invidious instances have the same number of users. In reality this will likely not be the case. A Pareto distribution is more probable. We can say that 80% of all Invidious users use 20% of all Invidious instances. The official instance at https://invidio.us/ is probably number one. So even though the official instance might have more computing power, it still might not be enough to handle all the traffic.

**Comment by mendel5** [comment]
@unixfox How do you know it's about the bots? I can think of two reasons for why the official Invidious instance ( https://invidio.us/ ) is not as performant as yours ( https://yewtu.be/ ).

1. Available computing resources
If we assume that all Invidious instances have the same number of users/ amount of load, then the provisioned computing resources can limit the performance. Maybe your instance has more CPU power, RAM, etc. than other instances?

2. Number of users
In (1) I assumed that all Invidious instances have the same number of users. In reality this will likely not be the case. A Pareto distribution is more probable. We can say that 80% of all Invidious users use 20% of all Invidious instances. The official instance at https://invidio.us/ is probably number one. So even though the official instance might have more computing power, it still might not be enough to handle all the traffic.

**Comment by unixfox** [comment]
A couple of months after launching my instance I quickly discovered that my postgresql database was growing like insane. Around 30GB then I cleaned it and again two weeks after that it was back at 30GB.
I looked at the logs and there were a bunch of clients abusing the API of invidious. I created an issue about that here: #946. 
Around that time my instance was running very slow.
Then I deployed a less intrusive anti bot solution (called filtron which doesn't requires JavaScript), after that I saw that my instance was back to normal and I looked at the logs and the requests that I though were bots was in reality real bots disguising as "real browsers" by setting their user agent to Chrome/Firefox.
And now one week ago the same scenario happened again. My instance was running quite slow and there were a bunch of requests of bots.
I decided to act by "upgrading" my anti bot solution and just like when before when I managed to stop the bots my instance went to a normal state.
That's why I have a feeling that the bots are ruining the public instances.

**Comment by mendel5** [comment]
@unixfox Very interesting! So the reason most public instances become slower is not the load itself that is created by the bots (e.g. CPU or RAM limit), but the database that becomes very large? I am trying to understand what causes the bottleneck.

**Comment by Perflyst** [comment]
> but the database that becomes very large

I dont think so. You can easily cleanup the database periodically and also optimize your postgresql configuration for better caching.
